### PR TITLE
Upgrade to Play 2.5 and Silhouette 4

### LINKED ISSUE #3103: Towards Play 2.6: Migrate Streaming Requests
part of #1622 
https://www.playframework.com/documentation/2.6.x/StreamsMigration25

### LINKED ISSUE #3151: Refactor Backend: find generic place for Request-To-AccessContext conversions
in #3136 we introduce some code duplication by having 
```scala
implicit def userAwareRequestToDBAccess(implicit request: sil.UserAwareRequest[_]) = DBAccessContext(request.identity)
implicit def securedRequestToDBAccess(implicit request: sil.SecuredRequest[_]) = DBAccessContext(Some(request.identity))
```
in every controller. This used to be in a trait, but since silhouette is injected now, that doesn’t work anymore.
Can we somehow find a generic solution again?

### LINKED ISSUE #3159: Towards Play 2.6: upgrade to silhouette 4.0
related:
- https://stackoverflow.com/questions/42209678/how-to-implement-multiple-silhouette-authenticators
- https://www.silhouette.rocks/v4.0/docs/migration-guide
- https://github.com/mohiva/play-silhouette-seed
- https://stackoverflow.com/questions/40472317/play-2-5-silhouette-4-di-with-guice

### PR DESCRIPTION
**Pull request by fm3**
This now consists of the changes to our uses of the stream API (originally committed here) as well as the PRs on logging (#3168), Silhouette 4 (#3160), content-disposition (#3171)

Users will be logged out on deploy, as we use new cookies. Rollback should use the old cookies again, roll-forward after that, the installed new ones.

### URL of deployed dev instance (used for testing):
- https://playstreams.webknossos.xyz

### Steps to test:
- all of the backend routes should be tested
- special focus on IO like uploading/downloading zips/nmls
- login should still work
- create a project, trace and finish some tasks, view it, download it
- [standalone datastore setup](https://github.com/scalableminds/webknossos/wiki/Set-up-a-standalone-datastore-locally) should also still work (not block data requests because CORS etc)

### Issues:
- fixes #3103
- fixes #3159
- fixes #3151 
- fixes #3103

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by fm3** [comment]
In this branch, the datastore compiles + works, and wK compiles but does not start (due to play.current, which is currently being tackled in #3136  #3142 and #3144).

However, wK compiles only while
 https://github.com/scalableminds/webknossos/pull/3125/files#diff-6facac230d118b7c3b0e41668bf0ade6R149
is commented out, because our Zip helper functions still expect a `NamedEnumeratorStream[Array[Byte]]` but the fetched data from the datastore is now received as Akka’s `Source[ByteStream, _]`.
This needs to be sorted out. I could not yet find any conversion methods between the two.

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala [hunk 1]
```diff
@@ -176,9 +179,10 @@ class BinaryDataController @Inject()(
             DataServiceRequestSettings(halfByte = halfByte))
           imageProvider <- respondWithSpriteSheet(dataSource, dataLayer, request, 1, blackAndWhite)
         } yield {
-          Ok.stream(Enumerator.outputStream(imageProvider).andThen(Enumerator.eof)).withHeaders(
-            CONTENT_TYPE -> contentTypeJpeg,
-            CONTENT_DISPOSITION -> "filename=test.jpg")
+          Result(
+            header = ResponseHeader(200),
+            body = HttpEntity.Streamed(StreamConverters.asOutputStream().mapMaterializedValue { outputStream => imageProvider(outputStream) }, None, Some(contentTypeJpeg)))
+          //TODO: disposition
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala]: Not sure if we even need the content-disposition, but then we can remove the TODO

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala [hunk 2]
```diff
@@ -200,9 +204,10 @@ class BinaryDataController @Inject()(
         for {
           thumbnailProvider <- respondWithImageThumbnail(dataSetName, dataLayerName, width, height, centerX, centerY, centerZ, zoom)
         } yield {
-          Ok.stream(Enumerator.outputStream(thumbnailProvider).andThen(Enumerator.eof)).withHeaders(
-            CONTENT_TYPE -> contentTypeJpeg,
-            CONTENT_DISPOSITION -> "filename=thumbnail.jpg")
+          Result(
+            header = ResponseHeader(200),
+            body = HttpEntity.Streamed(StreamConverters.asOutputStream().mapMaterializedValue { outputStream => thumbnailProvider(outputStream) }, None, Some(contentTypeJpeg)))
+          //TODO: disposition
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala]: same as above

**Comment by fm3** [comment]
@jstriebel do you want to upgrade the migration guide? I believe the only thing is the logback files. And I’m not entirely clear about what needs to be changed in deployments for that

**Pull request merged**